app/services/publishers/job_application_sort.rb

```diff
@@ -1,13 +1,15 @@
 class Publishers::JobApplicationSort < RecordSort
-  def initialize
-    @column = options.first.column
-    @order = options.first.order
+  def options
+    [submitted_at_desc_option, last_name_option]
   end
 
-  def options
-    [
-      SortOption.new("submitted_at", "desc", I18n.t("publishers.vacancies.job_applications.index.sort_by.date_received")),
-      SortOption.new("last_name", "asc", I18n.t("publishers.vacancies.job_applications.index.sort_by.applicant_last_name")),
-    ]
+  private
+
+  def last_name_option
+    @last_name_option ||= SortOption.new("last_name", I18n.t("publishers.vacancies.job_applications.index.sort_by.applicant_last_name"))
```

Comment:
I reckon this isn't worth memoizing - it's not all that expensive to compute (ditto below).